lib/util/transformer/length-based-splitter.js

```diff
@@ -96,6 +118,39 @@ export class LengthBasedSplitter extends Stream.Transform {
     this.isShutdown = true;
   }
 
+  /**
+   * Log information about large frames for debugging purposes
+   * @param {number} messageLength - The size of the detected frame
+   */
+  _logLargeFrame(messageLength) {
+    log.warn(`Large frame detected: ${util.toReadableSizeString(messageLength)}`);
+    log.warn(`Service: ${this.serviceName}, Max allowed: ${util.toReadableSizeString(this.maxFrameLength)}`);
+    
+    // Log memory usage if monitoring is enabled
+    if (DEBUG_CONFIG.enableMemoryMonitoring) {
+      const memUsage = process.memoryUsage();
+      log.warn(`Memory usage - RSS: ${util.toReadableSizeString(memUsage.rss)}, Heap: ${util.toReadableSizeString(memUsage.heapUsed)}`);
+    }
+  }
+
+  /**
+   * Handle frame size errors with enhanced debugging information
+   * @param {number} messageLength - The size of the frame that exceeded the limit
+   */
+  _handleFrameSizeError(messageLength) {
+    const errorMsg = `The frame is bigger than expected. Length: ${util.toReadableSizeString(messageLength)}, max: ${util.toReadableSizeString(this.maxFrameLength)}`;
+    log.error(errorMsg);
+    log.error(`Service: ${this.serviceName}, Frame offset: ${this.lengthFieldOffset}, Frame length: ${this.lengthFieldLength}`);
+    
+    // Suggest debugging steps
+    log.error('To debug this issue, set environment variables:');
+    log.error('  APPIUM_IOS_DEBUG_FRAME_SIZE=true');
+    log.error('  APPIUM_IOS_LOG_LARGE_FRAMES=true');
+    log.error('  APPIUM_IOS_MONITOR_MEMORY=true');
+    
+    throw new Error(errorMsg);
```

Comment:
it would be easier for the compiler to understand the code flow if this function returns the error object instead of throwing it, e.g. replace:

```
this._handleFrameSizeError(messageLength);
```

with 

```
throw this._makeFrameSizeError(messageLength);
```

and change the last expression in the method to 

```
return new Error(errorMsg);
```